### PR TITLE
feat: Add DeleteColumnReader and PositionColumnReader

### PR DESCRIPTION
**Pull request by huaxingao**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Add `DeleteColumnReader` and `PositionColumnReader` for iceberg

### CODE REVIEW

File: common/src/main/java/org/apache/comet/parquet/PositionColumnReader.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    this.position = position;
+  }
+
+  @Override
+  public void readBatch(int total) {
+    Native.resetBatch(nativeHandle);
+    Native.setPosition(nativeHandle, position, total);
+    position += total;
+
+    super.readBatch(total);
```
Comment (andygrove) [inline common/src/main/java/org/apache/comet/parquet/PositionColumnReader.java:45]: I'm not familiar with this area of code, but this looks odd to me and maybe just needs a comment to explain. My understanding:

- we set the position to `position`
- we then increase the `position` by `total`
- then we do the read, but shouldn't this be from the original position?

File: common/src/main/java/org/apache/comet/parquet/PositionColumnReader.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.comet.parquet;
+
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.spark.sql.types.DataTypes;
+
+/** A column reader for reading position column */
```
Comment (viirya) [inline common/src/main/java/org/apache/comet/parquet/PositionColumnReader.java:25]: Would you like to add some more comment explaining why we have these? Especially they are not used in Comet itself.

Comment (viirya) [reply common/src/main/java/org/apache/comet/parquet/PositionColumnReader.java:25]: Actually this is `common` module, and it is odd to have Iceberg code here.

Comment (huaxingao) [reply common/src/main/java/org/apache/comet/parquet/PositionColumnReader.java:25]: These are pretty much similar to `ConstantColumnReader`, so I am thinking of putting them in `common`.

Comment (huaxingao) [reply common/src/main/java/org/apache/comet/parquet/PositionColumnReader.java:25]: I am OK with leave these in Iceberg.

**Pull request closed without merging**